### DISCUSSION TITLE
window.__TAURI__ undefined on open local webserver URL

### DISCUSSION THREAD

**Discussion by igorjacauna** [tauri-apps/tauri#4828] [Q&A] [answered]
I run, as sidecar, a rust web server on port 7878 that works fine delivering static HTML.

And I have in my `tauri.config.json`, `withGlobalTauri: true`:

```json
{
  "$schema": "../node_modules/@tauri-apps/cli/schema.json",
  "build": {
    "beforeBuildCommand": "yarn build",
    "beforeDevCommand": "yarn dev",
    "devPath": "http://localhost:3000",
    "distDir": "../dist",
    "withGlobalTauri": true
  },
  ...
}
```

But when I open a URL from sidecar rust web server, `window.__TAURI__` and all others `window.*` Tauri APIs are `undefined`

I open with:

```ts
const localServer = new WebviewWindow('local-rust-web-server', {
  url: 'http://localhost:7878/',
});
```

And if try use `window.__TAURI__` on HTML that are delivered by rust web server it is undefined

**Comment by FabianLars** [comment, marked as answer]
This is expected behavior. Tauri's apis won't be injected into windows with external-ish urls
_Edit:_ See [here](https://togithub.com/tauri-apps/tauri-plugin-localhost/issues/12) for a similar issue (that got resolved in a hacky way)
